src/regtech_data_validator/create_schemas.py

```diff
@@ -114,8 +114,9 @@ def validate(schema: DataFrameSchema, submission_df: pd.DataFrame, max_errors: i
         # NOTE: `type: ignore` because SchemaErrors.schema_errors is supposed to be
         #       `list[dict[str,Any]]`, but it's actually of type `SchemaError`
         schema_error: SchemaError
-        single_field, multi_field, register = get_scope_counts(err.schema_errors)
-        total_error_count = sum([single_field, multi_field, register])
+        error_counts = get_scope_counts(err.schema_errors, Severity.ERROR)
+        warning_counts = get_scope_counts(err.schema_errors, Severity.WARNING)
```

Comment:
feels like this can be refactored so we're only processing through the errors once instead of looping through the same set again.

In `get_scope_counts` we can have single, multi, and register variables still be the list of the specific types of errors, instead of directly being the count; then construct the `Count` by filtering on the lists of the original implementation with severity, and returning the tuple of `error_counts`, and `warning_counts`

Reply:
@lchen-2101 I think I get what you're saying, and I thought about that after I wrote this story, but the original impl is a list of sums, not a list of checks that can be further filtered.  Are you saying instead of:
```
    single = [
        (~error.check_output).sum()
        for error in schema_errors
        if isinstance(error.check, SBLCheck) and error.check.scope == 'single-field'
    ]
```
do something like:
```
    singles = [
        error for error in schema_errors
        if isinstance(error.check, SBLCheck) and error.check.scope == 'single-field'
    ]
   single_errors = [(~error.check_output).sum() for error in singles if error.severity == Severity.ERROR]
   single_warnings = [(~error.check_output).sum() for error in singles if error.severity == Severity.WARNING]
   ...

   return Counts(...), Counts(...)
```
?

Reply:
yeap, exactly

Reply:
oh right... the original implementation was sum, not the errors themselves, but yeah, what you have above is what I was thinking

Reply:
Groovy, @guffee23 reach out if you have questions.  Since the whole point of this is to gracefully handle really large datasets, not processing a really large set of errors twice is probably a good thing lol.